File: aoc2021/day11/src/main/java/de/havox_design/aoc2016/day11/Day11.java
package de.havox_design.aoc2016.day11;

import de.havox_design.aoc.utils.java.input.DataReader;

import java.util.List;

public class Day11 {
    private final List<String> input;

    public Day11(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day11 instance = new Day11(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day11 instance = new Day11(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}

File: templates/scala/aoc20xx-scala/src/main/scala/de/havox_design/aoc20xx/day11/Day11.scala
package de.havox_design.aoc2022.day11

object Day11 {
  def solvePart1(filename: String): Long = {
    1L
  }

  def solvePart2(filename: String): Long = {
    1L
  }

  def main(args: Array[String]): Unit = {
    println("Solution for part1: " + solvePart1("de/havox_design/aoc20xx/day11/day11.txt"))
    println("Solution for part2: " + solvePart2("de/havox_design/aoc20xx/day11/day11.txt"))
  }

  private def readData(filename: String): Iterator[String] =
    scala
      .io
      .Source
      .fromResource(filename)
      .getLines()
}
